issue_2/test/TestGreenVault.ts

```diff
@@ -0,0 +1,136 @@
+import hre from "hardhat";
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/dist/src/signer-with-address";
+
+import { expect } from "chai";
+import { GreenToken } from "../typechain/GreenToken";
+import { GreenVault } from "../typechain/GreenVault";
+
+const { deployContract } = hre.waffle;
+
+declare module "mocha" {
+  export interface Context {
+    gtoken: GreenToken;
+    another_token: GreenToken;
+
+    gvault: GreenVault;
+    signers: { admin: SignerWithAddress, user1: SignerWithAddress, user2: SignerWithAddress };
+  }
+}
+
+describe("Unit tests", function () {
+  before(async function () {
+    const signers = await hre.ethers.getSigners();
+    this.signers = {
+      admin: signers[0],
+      user1: signers[1],
+      user2: signers[2],
+    }
+  });
+
+  describe("GreenVault", function () {
+    const starting_balance = 1337;
+
+    beforeEach(async function () {
+      this.gtoken = <GreenToken>await deployContract(this.signers.admin,
+        await hre.artifacts.readArtifact("GreenToken"), []);
+
+      this.another_token = <GreenToken>await deployContract(this.signers.admin,
+        await hre.artifacts.readArtifact("GreenToken"), []);
+
+      this.gvault = <GreenVault>await deployContract(this.signers.admin,
+        await hre.artifacts.readArtifact("GreenVault"), []);
+
+      for (const user of [this.signers.user1, this.signers.user2]) {
+        await this.gtoken.connect(user).mint(user.address, starting_balance);
+      }
+    });
+
+    it("starts empty", async function () {
+      expect(await this.gvault.connect(this.signers.user1).deposits(this.signers.user1.address, this.gtoken.address))
```

Comment:
I would probably use some variables to make the tests more readable:
```
const gvault = this.gvault.connect(this.signers.user1)
const user1 = this.signers.user1.address
```

Reply:
good point. connected gvault to user1 by default